src/display-buffer.coffee

```diff
@@ -41,6 +41,12 @@ class DisplayBuffer extends Model
     @disposables.add @tokenizedBuffer.onDidChange @handleTokenizedBufferChange
     @disposables.add @buffer.onDidCreateMarker @handleBufferMarkerCreated
     @foldMarkerAttributes = Object.freeze({class: 'fold', displayBufferId: @id})
+    if @largeFileMode
+      @tokenizedBuffer.onDidLoad(=> @finalizeConstruction())
```

Comment:
I definitely think it will be. This is a temporary stopgap, although we may not ever initialize smaller files in a background process because it might not be worth the overhead.